File: src/Front/src/api/impl/DictionaryFieldInfo.ts
// tslint:disable
// TypeScriptContractGenerator's generated content
import { FieldType } from './FieldType';
import { FieldInfo } from './FieldInfo';
import { FieldMeta } from './FieldMeta';

export type DictionaryFieldInfo = {
    type: FieldType.Dictionary;
    key: Nullable<FieldInfo>;
    value: Nullable<FieldInfo>;
    meta: Nullable<FieldMeta>;
};

File: src/Core/TypeInformation/DictionaryFieldInfo.cs
namespace Kontur.DBViewer.Core.TypeInformation
{
    public class DictionaryFieldInfo : FieldInfo
    {
        public DictionaryFieldInfo(FieldInfo key, FieldInfo value)
        {
            Key = key;
            Value = value;
        }

        public DictionaryFieldInfo()
        {
        }

        public override FieldType Type => FieldType.Dictionary;
        public FieldInfo Key { get; set; }
        public FieldInfo Value { get; set; }
    }
}